// langsmith/dist/_openapi_client/resources/runs/runs.cjs
"use strict";
// @ts-nocheck
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Runs = void 0;
const resource_js_1 = require("../../core/resource.cjs");
const ShareAPI = __importStar(require("./share.cjs"));
const share_js_1 = require("./share.cjs");
const pagination_js_1 = require("../../core/pagination.cjs");
const headers_js_1 = require("../../internal/headers.cjs");
const path_js_1 = require("../../internal/utils/path.cjs");
class Runs extends resource_js_1.APIResource {
    constructor() {
        super(...arguments);
        Object.defineProperty(this, "share", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: new ShareAPI.Share(this._client)
        });
        Object.defineProperty(this, "retrieve", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: this.retrieveV2
        });
        Object.defineProperty(this, "query", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: this.queryV2
        });
    }
    /**
     * Returns the URL to view a specific run in the LangSmith UI. The caller must
     * supply the run's project_id and trace_id as query parameters; start_time is
     * optional.
     *
     * @example
     * ```ts
     * const response = await client.runs.getURL(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { project_id: 'project_id', trace_id: 'trace_id' },
     * );
     * ```
     */
    getURL(runID, query, options) {
        return this._client.get((0, path_js_1.path) `/v2/runs/${runID}/url`, { query, ...options });
    }
    /**
     * **Alpha:** The request and response contract may change; Returns a paginated
     * list of runs for the given projects within min/max start_time. Supports filters,
     * cursor pagination, and `selects` to select fields to return.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const run of client.runs.queryV2()) {
     *   // ...
     * }
     * ```
     */
    queryV2(params, options) {
        const { Accept, ...body } = params;
        return this._client.getAPIList('/v2/runs/query', (pagination_js_1.ItemsCursorPostPagination), {
            body,
            method: 'post',
            ...options,
            headers: (0, headers_js_1.buildHeaders)([{ ...(Accept != null ? { Accept: Accept } : undefined) }, options?.headers]),
        });
    }
    /**
     * **Alpha:** The request and response contract may change; Returns one run by ID
     * for the given session. Use the `selects` query parameter (repeatable) to select
     * fields to return.
     *
     * @example
     * ```ts
     * const run = await client.runs.retrieveV2(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { project_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    retrieveV2(runID, params, options) {
        const { Accept, ...query } = params;
        return this._client.get((0, path_js_1.path) `/v2/runs/${runID}`, {
            query,
            ...options,
            headers: (0, headers_js_1.buildHeaders)([{ ...(Accept != null ? { Accept: Accept } : undefined) }, options?.headers]),
        });
    }
}
exports.Runs = Runs;
Runs.Share = share_js_1.Share;
